frontend/src/components/groups/MembersTableUserEntry.tsx

```diff
@@ -61,74 +62,102 @@ export function MembersTableUserEntry(props: MembersTableUserEntryProps) {
 		setEditRoleOn(false);
 	};
 
-	return (
-		<TableRow
-			key={member.user.id}
-			sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
-		>
-			<TableCell component="th" scope="row" key={`${member.user.id}-icon`}>
-				{
-					member && member.user && member.user.email ?
-						<Gravatar email={member.user.email} rating="g"
-							  style={{width: "32px", height: "32px", borderRadius: "50%", verticalAlign: "middle"}}/>
-						:
-						<PersonIcon sx={iconStyle}/>
-				}
-				<Button >{member.user.first_name} {member.user.last_name}</Button>
-			</TableCell>
-			<TableCell align="right">{member.user.email}</TableCell>
-			<TableCell align="right">
-				{
-					editRoleOn ?
-						<FormControl size="small">
-							  <InputLabel id="demo-simple-select-label">Role</InputLabel>
-							  <Select
-								labelId="role"
-								id="role"
-								value={selectedRole}
-								label="Role"
-								onChange={handleRoleSelection}
-							  >
-								<MenuItem value={"editor"}>Editor</MenuItem>
-								<MenuItem value={"member"}>Member</MenuItem>
-							  </Select>
-							</FormControl>
-						:
-						member.editor !== undefined && member.editor ?
-							"Editor" : "Member"
-				}
-				{/*only owner or editor are allowed to modify roles of the member*/}
-				<AuthWrapper currRole={role} allowedRoles={["owner", "editor"]}>
+	console.log(member, 'member')
+
+	if (member.user.email == creatorEmail) {
```

Comment:
Since all we are doing is not letting the user change this, can we put the if statement around only those controls and not the entire row? This way we don't have to duplicate as much code?

Reply:
Put each TableCell into conditionals. This also makes it clearer what is different based on Owner.